cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/util/Misc.java

```diff
@@ -112,8 +113,8 @@ public static void checkNotNull(Object param, String errorMessagePrefix) throws
      */
     public static void checkNotNullOrEmpty(String param, String errorMessagePrefix) throws
             IllegalArgumentException {
-        checkArgument(!isStringNullOrEmpty(param), (errorMessagePrefix != null ? errorMessagePrefix :
-                "Parameter") + " must not be " + param == null ? "null." : "empty.");
+        // Null is never passed as the parameter here, as per FindBugs and Intellij inspections.
```

Comment:
Just double checked the usages, the front facing API uses this. First glance it was just the SQL code, so I removed the check, backing out and replacing with exclude.